packages/db-objects-redis/lib/objects/objectsInRedisClient.js

```diff
@@ -3969,6 +3993,41 @@ class ObjectsInRedisClient {
         }
         return noMigrated;
     }
+
+    /**
+     * Update the protocol version in DB
+     *
+     * @returns {Promise<void>}
+     */
+    async _updateProtocolVersion() {
+        await this.client.set(`${this.metaNamespace}dbProtocol.${tools.getHostName()}`, this.dbProtocolVersion);
+    }
+
+    /**
+     * Remove the protocol version from DB, has to be done if host gets deleted
+     *
+     * @returns {Promise<void>}
+     */
+    async deleteProtocolVersion() {
+        await this.client.del(`${this.metaNamespace}dbProtocol.${tools.getHostName()}`);
+    }
+
+    /**
+     * Evaluate if there are non supportive protocol versions active
+     * @returns {Promise<void>}
+     * @private
+     */
+    async _checkAllowedToStart() {
+        const keys = await this._getKeysViaScan(`${this.metaNamespace}dbProtocol.*`);
+
+        for (const key of keys) {
+            const version = this.client.get(key);
+            if (parseInt(version) > this.dbProtocolVersion) {
+                // in this environment there is a newer version, we are not allowed to start
+                throw new Error(`${key.split('.')[3]} has a a newer DB version which is incompatible`);
```

Comment:
makes that really sense to say "the highest host is the leader regarding version"? Because "higher hosts" are backward compatible ...

Thats why my idea is to have one more meta "effectiveProtocolversion" Field which defines the version to use. And in fact each host need to check if it can support this version ... then start ... else not

Reply:
my idea is only increase the protocolVersion if it contains a breaking change, and if so then the host will not be backward compatible because, e.g. the new host uses sets but the old host does not maintain sets to its objects it will be messed up.

Reply:
will be another pr